ui/base-tool/src/main/java/org/cru/godtools/base/tool/viewmodel/ToolStateHolder.kt

```diff
@@ -0,0 +1,13 @@
+package org.cru.godtools.base.tool.viewmodel
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import org.cru.godtools.tool.state.State
+
+private const val STATE_TOOL_STATE = "toolState"
+
+class ToolStateHolder(private val savedState: SavedStateHandle) : ViewModel() {
+    val toolState by lazy {
```

Comment:
So will these quizzes be in multiple parts of the app, hence why you added the tool state to different activities, controllers, etc? And each will instantiate its own individual state.

Reply:
yes, `ToolState` is a new concept that extends beyond just quizzes. it will allow content elements to store arbitrary data and other content elements to change their behavior on what data is stored.

The scope of `ToolState` is across the entirety of the a single tool during a session using it. So state stored on the first page of a tool will be able to be referenced on the last page of that same tool. But if the user backs out of a tool and re-enters it the state will be reset.